frontend/src/components/dialogs/SecretDialogOpenstack.vue

```diff
@@ -293,6 +318,23 @@ export default {
         return 'OpenStack Designate'
       }
       return undefined
+    },
+    authenticationMethod: {
+      get () {
+        return this.authenticationMethodValue
+      },
+      set (value) {
+        this.authenticationMethodValue = value
+        if (value === 'user') {
+          this.username = undefined
+          this.password = undefined
+        }
+        if (value === 'appCredentials') {
+          this.applicationCredentialID = undefined
+          this.applicationCredentialName = undefined
+          this.applicationCredentialSecret = undefined
+        }
```

Comment:
why not simply reset all username, password and application credentials when the authenticationMethod changes? Or is there a reason for it to be conditional?